Replace Celo Gold with CELO and add warning about using Ethereum wallets (#4157)
celo-org/celo-monorepo: packages/docs/GLOSSARY.md

An open platform that makes financial tools accessible to anyone with a mobile phone.

## CELO

The ticker symbol for the [Celo native asset](#celo-native-asset), often written in capital letters to avoid confusion with references to the Celo protocol.

## Celo Gold

> The deprecated name for the Celo native asset, which now is referred to simply as "Celo" or "CELO" when it might be ambiguous whether the reference is to the asset or the protocol.

Reviewer: ```suggestion
The deprecated name for the Celo native asset, which now is referred to simply as "Celo" or preferably "CELO".
```
